### ISSUE TITLE
Improving the text compare feature

### ISSUE THREAD

**Issue by q2apro** [svivian/q2a-edit-history#22]
Example of recent version:

![image](https://user-images.githubusercontent.com/6764548/42439910-d88b52e2-836c-11e8-831b-8fa1c379112c.png)

The plugin also considers line breaks (not the HTML ones, but ENTER key, next line) as change. I suggest to ignore those, to get a better overview.

**Comment by svivian** [comment]
Fair point, I'll see what i can do. Is there a reason the line breaks were changed? Is that something to do with the editor?

**Comment by q2apro** [comment]
Yes. As far as I could see, the sceditor formats the HTML content string. Doing indents and line breaks. See the plugin in my repo if you like to test.

**Comment by q2apro** [comment]
I would add a compare option:

- HTML
- text only
- ... (Other modes)

**Comment by q2apro** [comment]
Stackoverflow uses 3 compare modes, example: https://meta.stackoverflow.com/posts/256092/revisions

**Comment by q2apro** [comment]
If you do a `strip_tags` on both content elements, the result looks like:

![image](https://user-images.githubusercontent.com/6764548/43567299-f50488a0-9639-11e8-8d51-ce787060295a.png)

By the way, I found out that SCEditor is doing the HTML reformating of my old posts, when an edit happens. Then the HTML tags are (backend) moved to other positions. That's why the plugin displays those changes.

**Comment by q2apro** [comment]
I think a solution could be to remove all line breaks and then compare!

	// q2apro - need if HTML formating changes, but text content is the same
	function q2apro_remove_linebreaks($string)
	{
		return preg_replace( "/\r|\n/", "", $string);
	}

And then:

	$rev['content'] = $this->q2apro_remove_linebreaks($rev['content']);
	// ...
	$revisions[0]['diff_content'] = $this->q2apro_remove_linebreaks($revisions[0]['diff_content']);

Result looks good:

![image](https://user-images.githubusercontent.com/6764548/43567507-abd31bd2-963a-11e8-95c8-76f8c13e5e19.png)

It should be checked if the post format is `html`, then remove line breaks. If the post format is `text`, then do not. 👍

**Comment by q2apro** [comment]
Another solution would be to remove all tabs and spaces from the beginning of each line.

But I am not sure, what is best.

The solution above is good for small texts. If it is a longer text, all is on one long line.

**Comment by q2apro** [comment]
Example for comparision:

**Revision with recent plugin:**
![image](https://user-images.githubusercontent.com/6764548/43568262-c968451c-963c-11e8-9868-d54ba6f70c1b.png)

**Revision with Mod "q2apro_remove_linebreaks"**
![image](https://user-images.githubusercontent.com/6764548/43568280-dbfbb88a-963c-11e8-94c0-bcef5ccb1ba6.png)

**Revision with Mod "q2apro_strip_tags_with_space"**
![image](https://user-images.githubusercontent.com/6764548/43568330-fd5cde28-963c-11e8-9c9c-a5f34f29a037.png)

	// q2apro - need if HTML formating changes, but text content is the same
	function q2apro_strip_tags_with_space($string)
	{
		return strip_tags(str_replace('<', ' <', $string));
	}
	
The last one is in my opinion the easiest to read.
